sgt-back-end/index.js

```diff
@@ -0,0 +1,195 @@
+  from "grades"
+  `;
+
+  db.query(sql)
+    .then(result => {
+      res.status(200).json(result.rows);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occured.'
+      });
+    });
+});
+
+app.post('/api/grades', (req, res) => {
+  const entry = req.body;
+  if (!entry.name || !entry.course) {
+    res.status(400).json({
+      error: 'Please enter valid name and/or course'
+    });
+  } else if (!Number.isInteger(Number(entry.score)) || entry.score <= 0) {
+    res.status(400).json({
+      error: 'Please enter valid score'
+    });
+  } else {
+    const sql = `
+  insert into "grades" ("name", "course", "score")
+  values ($1,$2,$3)
+  returning *
+  `;
+    const params = [entry.name, entry.course, entry.score];
+    db.query(sql, params)
+      .then(result => {
+        res.status(201).json(entry);
+      })
+      .catch(err => {
+        console.error(err);
+        res.status(500).json({
+          error: 'An unexpected error occured'
+        });
+      });
+  }
+});
+
+app.put('/api/grades/:id', (req, res) => {
+  const id = Number(req.params.id);
+  const entry = req.body;
+  if (!Number.isInteger(id) || id <= 0) {
+    res.status(400).send({
+      error: 'id must be a positive integer'
+    });
+    return;
+  }
+  if (!entry.name || !entry.course) {
+    res.status(400).json({
+      error: 'Please enter valid name and/or course'
+    });
+  } else if (!Number.isInteger(Number(entry.score)) || entry.score <= 0) {
+    res.status(400).json({
+      error: 'Please enter valid score'
+    });
+  } else {
+    const sql = `
+    update "grades"
+    set "name" = $1,
+        "course" = $2,
+        "score" = $3
+        where "gradeId" = $4
+        returning *
+        `;
+    const params = [entry.name, entry.course, entry.score, id];
+    db.query(sql, params)
+      .then(result => {
+        if (!result.rows[0]) {
+          res.status(404).json({
+            error: `Cannot find a record matching ${id}`
+          });
+        }
+        res.status(200).json(entry);
+      })
+      .catch(err => {
+        console.error(err);
+        res.status(500).json({
+          error: 'An unexpected error occured'
+        });
+      });
+  }
+});
+
+app.delete('/api/grades/:id', (req, res) => {
+  const id = Number(req.params.id);
+  if (!Number.isInteger(id) || id <= 0) {
+    res.status(400).json({
+      error: 'id must be a positive interger'
+    });
+    return;
+  }
+  if (!id) {
+    res.status(404).json({
+      error: 'An unexpected error occured'
+    });
+    return;
+  }
+  const sql = `
+  delete from "grades"
+  where "gradeId" = $1
+  returning *
+  `;
+  const params = [id];
+  db.query(sql, params)
+    .then(result => {
+      if (!result.rows[0]) {
+        res.status(404).json({
+          error: `Cannot find a record matching ${id}`
+        });
+      }
+      res.status(204).json({
+        success: 'deleted entry'
+      });
```

Comment:
With status `204` we can't send back any data. Status `204` it means "No Content" so you can't send a response `body`.  So it is best to use `sendStatus`
```JAVASCRIPT
res.sendStatus(204);
```